src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleLogFormatter.cs

```diff
@@ -0,0 +1,160 @@
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class JsonConsoleLogFormatter : ConsoleFormatter, IDisposable
+    {
+        private IDisposable _optionsReloadToken;
+
+        public JsonConsoleLogFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options)
+            : base (ConsoleFormatterNames.Json)
+        {
+            FormatterOptions = options.CurrentValue;
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            LogLevel logLevel = logEntry.LogLevel;
+            string category = logEntry.Category;
+            int eventId = logEntry.EventId.Id;
+            string message = logEntry.Formatter(logEntry.State, logEntry.Exception);
+            Exception exception = logEntry.Exception;
+            const int DefaultBufferSize = 1024;
+            var output = new ArrayBufferWriter<byte>(DefaultBufferSize);
+            using (var writer = new Utf8JsonWriter(output, FormatterOptions.JsonWriterOptions))
+            {
+                writer.WriteStartObject();
+                string timestamp = null;
+                var timestampFormat = FormatterOptions.TimestampFormat;
+                if (timestampFormat != null)
+                {
+                    var dateTime = FormatterOptions.UseUtcTimestamp ? DateTime.UtcNow : DateTime.Now;
+                    timestamp = dateTime.ToString(timestampFormat);
+                }
+                writer.WriteString("Timestamp", timestamp);
+                writer.WriteNumber(nameof(logEntry.EventId), eventId);
+                writer.WriteString(nameof(logEntry.LogLevel), GetLogLevelString(logLevel));
+                writer.WriteString(nameof(logEntry.Category), category);
+                writer.WriteString("Message", message);
+
+                if (exception != null)
+                {
+                    writer.WriteStartObject(nameof(Exception));
+                    writer.WriteString(nameof(exception.Message), exception.Message.ToString());
+                    writer.WriteString("Type", exception.GetType().ToString());
+                    writer.WriteStartArray(nameof(exception.StackTrace));
+                    string stackTrace = exception?.StackTrace;
+                    if (stackTrace != null)
+                    {
+                        foreach (var stackTraceLines in stackTrace?.Split(Environment.NewLine))
```

Comment:
Why are we doing this?

Reply:
refer to https://github.com/dotnet/runtime/pull/38616#discussion_r451212539